### PR TITLE
plan: resolve table source by infoschema.

### PR DESCRIPTION
**Pull request by hanfei1991**
We will remove resolver.go
@shenli @zimulala @coocood PTAL

### CODE REVIEW

**Comment by coocood** [comment]
LGTM

**Comment by shenli** [comment]
Why we should remove  resolver.go?

**Comment by hanfei1991** [comment]
@shenli we can do resolving during plan builder phase.

**Comment by hanfei1991** [comment]
@shenli @zimulala PTAL

File: plan/logical_plan_builder.go [hunk 1]
```diff
@@ -887,28 +887,47 @@ func (b *planBuilder) buildDataSource(tn *ast.TableName) LogicalPlan {
...
+		if b.inUpdateStmt {
+			switch col.State {
+			case model.StatePublic, model.StateWriteOnly, model.StateWriteReorganization:
+			default:
+				continue
+			}
+		} else if col.State != model.StatePublic {
+			continue
+		}
+		p.DBName = &schemaName
```
Comment (zimulala) [inline plan/logical_plan_builder.go]: This line can put out of `for`. It's enough to be assigned once.

**Comment by hanfei1991** [comment]
@zimulala PTAL

**Pull request merged**